## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :x: [more...](#f1ef0dbf-3707-4954-84ee-b4366be1f11f) |
| mint-fs.sh | :x: [more...](#843a8d37-aaae-4179-bac0-00e9bba27705) |
| mint-erasure.sh | :x: [more...](#51a2e428-3808-451d-a1c6-4006c4083607) |
| mint-dist-erasure.sh | :x: [more...](#1560fa58-6c75-445b-b68a-0c08ffbf9dc5) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#f36e4862-6a99-4be4-aa1a-df09be865a98) |
| mint-pools.sh | :x: [more...](#773e413f-4a3e-41be-89b8-2bf5b5a036dc) |
<a name="773e413f-4a3e-41be-89b8-2bf5b5a036dc"></a>
### 16195-ff06910/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.8:32743
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 035dcfe7d1cd:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 1 minutes and 1 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 16 seconds
(8/14) Running minio-go tests ... FAILED in 20 seconds
{
  "args": {
    "bucketName": "minio-go-test-thv6m4ndwrcdaukg",
    "objectName": "2ym9jww2saue9dvdse9jn4a5mcs9xu"
  },
  "duration": 649,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "GetObject()",
  "message": "CleanupBucket failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 35 seconds
{
  "name": "minio-java",
  "function": "removeBucket()",
  "duration": 25,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty. You must delete all versions in the bucket., bucketName = minio-java-test-11kodsg, objectName = null, resource = /minio-java-test-11kodsg, requestId = 172EEB76971A9152, hostId = 0b29f66c-eaba-43aa-880d-ff7d73139fb5)\nrequest={method=DELETE, url=http://15.15.15.8:32743/minio-java-test-11kodsg, headers=Host: 15.15.15.8:32743\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20221208T203006Z\nAuthorization: ██\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 172EEB76971A9152\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Thu, 08 Dec 2022 20:30:06 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 44 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"Versioning tests on a buckets\"",
  "duration": 6,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 31 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-840417f4-6167-4ff3-bb97-adec00b706bd",
    "object_name": "abc4269a-34e6-4861-9514-d716590ebd61-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "X-Amz-Meta-Testing": "value"
    },
    "content_type": "application/octet-stream"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-840417f4-6167-4ff3-bb97-adec00b706bd, request_id: 172EEB8805DF6D6D, host_id: 0b29f66c-eaba-43aa-880d-ff7d73139fb5, bucket_name: minio-py-test-840417f4-6167-4ff3-bb97-adec00b706bd",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 815, in _test_stat_object\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 675, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-840417f4-6167-4ff3-bb97-adec00b706bd, request_id: 172EEB8805DF6D6D, host_id: 0b29f66c-eaba-43aa-880d-ff7d73139fb5, bucket_name: minio-py-test-840417f4-6167-4ff3-bb97-adec00b706bd\n",
  "duration": 9526
}
(8/14) Running s3cmd tests ... done in 16 seconds
(9/14) Running s3select tests ... done in 5 seconds
(10/14) Running versioning tests ... FAILED in 2 minutes and 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-mbcy5je4tzelwm",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 232,
  "function": "/mint/build/versioning/delete.go:169: main.testDeleteObject",
  "message": "ListObjectVersions returned some entries but expected to return nothing",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="843a8d37-aaae-4179-bac0-00e9bba27705"></a>
### 16195-ff06910/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:31676
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 1559b83ab784:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 12 seconds
(8/14) Running minio-go tests ... FAILED in 15 seconds
{
  "args": {
    "bucketName": "minio-go-test-y3lzqy3qgm1hungu",
    "objectName": "wklgk3hfsbtgw14hbuwtn22b1g5yyh"
  },
  "duration": 191,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "GetObject()",
  "message": "CleanupBucket failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 14 seconds
{
  "name": "minio-java",
  "function": "removeBucket()",
  "duration": 10,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty. You must delete all versions in the bucket., bucketName = minio-java-test-3l9iuc9, objectName = null, resource = /minio-java-test-3l9iuc9, requestId = 172EEB6DC4D653BD, hostId = ed315c66-3452-45c3-a2cf-3a67da350552)\nrequest={method=DELETE, url=http://15.15.15.1:31676/minio-java-test-3l9iuc9, headers=Host: 15.15.15.1:31676\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20221208T202928Z\nAuthorization: ██\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 172EEB6DC4D653BD\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Thu, 08 Dec 2022 20:29:28 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 39 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"Versioning tests on a buckets\"",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 28 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-614526d4-6354-4d1b-89e9-aa16dc0f945e",
    "object_name": "527b0658-a428-419c-bd6e-a3aa789c7595-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "X-Amz-Meta-Testing": "value"
    },
    "content_type": "application/octet-stream"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-614526d4-6354-4d1b-89e9-aa16dc0f945e, request_id: 172EEB7D638A44EC, host_id: ed315c66-3452-45c3-a2cf-3a67da350552, bucket_name: minio-py-test-614526d4-6354-4d1b-89e9-aa16dc0f945e",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 815, in _test_stat_object\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 675, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-614526d4-6354-4d1b-89e9-aa16dc0f945e, request_id: 172EEB7D638A44EC, host_id: ed315c66-3452-45c3-a2cf-3a67da350552, bucket_name: minio-py-test-614526d4-6354-4d1b-89e9-aa16dc0f945e\n",
  "duration": 8941
}
(8/14) Running s3cmd tests ... done in 12 seconds
(9/14) Running s3select tests ... done in 2 seconds
(10/14) Running versioning tests ... FAILED in 2 minutes and 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-kvm2n5tbibevz2",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 35,
  "function": "/mint/build/versioning/delete.go:169: main.testDeleteObject",
  "message": "ListObjectVersions returned some entries but expected to return nothing",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="f1ef0dbf-3707-4954-84ee-b4366be1f11f"></a>
### 16195-ff06910/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.8:31735
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 43aa520a0c00:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 1 minutes and 1 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 16 seconds
(8/14) Running minio-go tests ... FAILED in 17 seconds
{
  "args": {
    "bucketName": "minio-go-test-gkvg6vw0sfulszjj",
    "objectName": "xefi5zgqai00s6ypzk2guqvtoui1ss"
  },
  "duration": 565,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "GetObject()",
  "message": "CleanupBucket failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 30 seconds
{
  "name": "minio-java",
  "function": "removeBucket()",
  "duration": 22,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty. You must delete all versions in the bucket., bucketName = minio-java-test-e3g7dq, objectName = null, resource = /minio-java-test-e3g7dq, requestId = 172EEB749A18D14A, hostId = b5d6ac17-14cc-43b9-ad84-d98cc89c3296)\nrequest={method=DELETE, url=http://15.15.15.8:31735/minio-java-test-e3g7dq, headers=Host: 15.15.15.8:31735\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20221208T202957Z\nAuthorization: ██\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 371\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 172EEB749A18D14A\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Thu, 08 Dec 2022 20:29:57 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 44 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"Versioning tests on a buckets\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 31 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-bc8ba9a3-148e-4d1f-b2c0-310be72f892c",
    "object_name": "3f1cf7f9-0850-4e77-91ce-fd73d21a5d40-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "X-Amz-Meta-Testing": "value"
    },
    "content_type": "application/octet-stream"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-bc8ba9a3-148e-4d1f-b2c0-310be72f892c, request_id: 172EEB8633391C65, host_id: b5d6ac17-14cc-43b9-ad84-d98cc89c3296, bucket_name: minio-py-test-bc8ba9a3-148e-4d1f-b2c0-310be72f892c",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 815, in _test_stat_object\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 675, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-bc8ba9a3-148e-4d1f-b2c0-310be72f892c, request_id: 172EEB8633391C65, host_id: b5d6ac17-14cc-43b9-ad84-d98cc89c3296, bucket_name: minio-py-test-bc8ba9a3-148e-4d1f-b2c0-310be72f892c\n",
  "duration": 9514
}
(8/14) Running s3cmd tests ... done in 16 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 2 minutes and 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-hm4c34a6na1zb5",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 210,
  "function": "/mint/build/versioning/delete.go:169: main.testDeleteObject",
  "message": "ListObjectVersions returned some entries but expected to return nothing",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="1560fa58-6c75-445b-b68a-0c08ffbf9dc5"></a>
### 16195-ff06910/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:31326
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp f704782ecea4:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 1 minutes and 0 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 15 seconds
(8/14) Running minio-go tests ... FAILED in 17 seconds
{
  "args": {
    "bucketName": "minio-go-test-kntpyklyndwsttqd",
    "objectName": "aezp4tehm0fuxf1e56s1xs30z0ekxt"
  },
  "duration": 549,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "GetObject()",
  "message": "CleanupBucket failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 27 seconds
{
  "name": "minio-java",
  "function": "removeBucket()",
  "duration": 17,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty. You must delete all versions in the bucket., bucketName = minio-java-test-30go7l0, objectName = null, resource = /minio-java-test-30go7l0, requestId = 172EEB731B68874C, hostId = 7c06aea3-7293-4b60-b933-c1688cc086a6)\nrequest={method=DELETE, url=http://15.15.15.1:31326/minio-java-test-30go7l0, headers=Host: 15.15.15.1:31326\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20221208T202951Z\nAuthorization: ██\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 172EEB731B68874C\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Thu, 08 Dec 2022 20:29:51 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 42 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"Versioning tests on a buckets\"",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 31 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-888a94f4-d7ee-4b59-a414-3716b14d120d",
    "object_name": "3474a959-f6c2-4c33-9afd-50f0da0b22fa-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "X-Amz-Meta-Testing": "value"
    },
    "content_type": "application/octet-stream"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-888a94f4-d7ee-4b59-a414-3716b14d120d, request_id: 172EEB8409737DB1, host_id: 7c06aea3-7293-4b60-b933-c1688cc086a6, bucket_name: minio-py-test-888a94f4-d7ee-4b59-a414-3716b14d120d",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 815, in _test_stat_object\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 675, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-888a94f4-d7ee-4b59-a414-3716b14d120d, request_id: 172EEB8409737DB1, host_id: 7c06aea3-7293-4b60-b933-c1688cc086a6, bucket_name: minio-py-test-888a94f4-d7ee-4b59-a414-3716b14d120d\n",
  "duration": 9773
}
(8/14) Running s3cmd tests ... done in 15 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 2 minutes and 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-y4c3abugl56dnf",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 201,
  "function": "/mint/build/versioning/delete.go:169: main.testDeleteObject",
  "message": "ListObjectVersions returned some entries but expected to return nothing",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="f36e4862-6a99-4be4-aa1a-df09be865a98"></a>
### 16195-ff06910/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:32380
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp e8b264d2887f:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 1 minutes and 0 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 16 seconds
(8/14) Running minio-go tests ... FAILED in 20 seconds
{
  "args": {
    "bucketName": "minio-go-test-shaw0dp3qy4zmcp1",
    "objectName": "urcgp01y54azmb3b1u0yiocet65kzt"
  },
  "duration": 602,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "GetObject()",
  "message": "CleanupBucket failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 27 seconds
{
  "name": "minio-java",
  "function": "removeBucket()",
  "duration": 15,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty. You must delete all versions in the bucket., bucketName = minio-java-test-nk5sbv, objectName = null, resource = /minio-java-test-nk5sbv, requestId = 172EEB74167407FB, hostId = 1b1211ec-1510-454a-9887-95ff97f626d7)\nrequest={method=DELETE, url=http://15.15.15.1:32380/minio-java-test-nk5sbv, headers=Host: 15.15.15.1:32380\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20221208T202955Z\nAuthorization: ██\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 371\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 172EEB74167407FB\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Thu, 08 Dec 2022 20:29:55 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 43 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"Versioning tests on a buckets\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 31 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-db070d91-d063-48e0-b2f4-a67dfe34a0f7",
    "object_name": "124b025a-0bf2-4dc6-9ad4-31e1a8c013d7-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "X-Amz-Meta-Testing": "value"
    },
    "content_type": "application/octet-stream"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-db070d91-d063-48e0-b2f4-a67dfe34a0f7, request_id: 172EEB8530B7F36B, host_id: 1b1211ec-1510-454a-9887-95ff97f626d7, bucket_name: minio-py-test-db070d91-d063-48e0-b2f4-a67dfe34a0f7",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 815, in _test_stat_object\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 675, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-db070d91-d063-48e0-b2f4-a67dfe34a0f7, request_id: 172EEB8530B7F36B, host_id: 1b1211ec-1510-454a-9887-95ff97f626d7, bucket_name: minio-py-test-db070d91-d063-48e0-b2f4-a67dfe34a0f7\n",
  "duration": 9424
}
(8/14) Running s3cmd tests ... done in 15 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 2 minutes and 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-s56tycenbk4lcr",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 207,
  "function": "/mint/build/versioning/delete.go:169: main.testDeleteObject",
  "message": "ListObjectVersions returned some entries but expected to return nothing",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="51a2e428-3808-451d-a1c6-4006c4083607"></a>
### 16195-ff06910/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.5:30639
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 87ecdada6aa3:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 6 seconds
(5/14) Running awscli tests ... done in 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 13 seconds
(8/14) Running minio-go tests ... FAILED in 16 seconds
{
  "args": {
    "bucketName": "minio-go-test-wkd0hmcyhoablpbs",
    "objectName": "bllnf5jf2nyfopbtagbu3f3aq4fkf3"
  },
  "duration": 245,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "GetObject()",
  "message": "CleanupBucket failed",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 17 seconds
{
  "name": "minio-java",
  "function": "getObjectRetention()",
  "duration": 39,
  "status": "FAIL",
  "error": "java.time.format.DateTimeParseException: Text '2022-12-09T20:29:34.7Z' could not be parsed at index 19 >>> [io.minio.Xml.unmarshal(Xml.java:55), io.minio.MinioAsyncClient.lambda$getObjectRetention$35(MinioAsyncClient.java:1738), java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642), java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506), java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073), io.minio.S3Base$1.onResponse(S3Base.java:559), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... FAILED in 41 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"Versioning tests on a buckets\"",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/14) Running minio-py tests ... FAILED in 29 seconds
{
  "name": "minio-py:test_stat_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-cd709ed5-d4df-44ef-91d5-e4d1dac90d18",
    "object_name": "90123dab-dfab-47b8-8a1d-0fa1ea98318b-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "X-Amz-Meta-Testing": "value"
    },
    "content_type": "application/octet-stream"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-cd709ed5-d4df-44ef-91d5-e4d1dac90d18, request_id: 172EEB7FCCDC2F81, host_id: a579f1ff-b9cc-493f-b260-e89fdb7b466b, bucket_name: minio-py-test-cd709ed5-d4df-44ef-91d5-e4d1dac90d18",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 825, in test_stat_object_version\n    _test_stat_object(log_entry, sse, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 815, in _test_stat_object\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 675, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty. You must delete all versions in the bucket., resource: /minio-py-test-cd709ed5-d4df-44ef-91d5-e4d1dac90d18, request_id: 172EEB7FCCDC2F81, host_id: a579f1ff-b9cc-493f-b260-e89fdb7b466b, bucket_name: minio-py-test-cd709ed5-d4df-44ef-91d5-e4d1dac90d18\n",
  "duration": 9763
}
(8/14) Running s3cmd tests ... done in 15 seconds
(9/14) Running s3select tests ... done in 2 seconds
(10/14) Running versioning tests ... FAILED in 2 minutes and 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-m9ef1pgu9wfpcw",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 46,
  "function": "/mint/build/versioning/delete.go:169: main.testDeleteObject",
  "message": "ListObjectVersions returned some entries but expected to return nothing",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/.github/workflows/vulncheck.yml
+++ b/.github/workflows/vulncheck.yml
@@ -16,7 +16,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.19.x
+          go-version: 1.19
           check-latest: true
       - name: Get official govulncheck
         run: go install golang.org/x/vuln/cmd/govulncheck@latest

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -1556,13 +1556,24 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 	defer NSUpdated(bucket, object)
 
 	storageDisks := er.getDisks()
-
-	//  Determine whether to mark object deleted for replication
-	var markDelete bool
+	versionFound := true
+	objInfo = ObjectInfo{VersionID: opts.VersionID} // version id needed in Delete API response.
+	goi, _, gerr := er.getObjectInfoAndQuorum(ctx, bucket, object, opts)
+	if gerr != nil && goi.Name == "" {
+		switch gerr.(type) {
+		case InsufficientReadQuorum:
+			return objInfo, InsufficientWriteQuorum{}
+		}
+		// For delete marker replication, versionID being replicated will not exist on disk
+		if opts.DeleteMarker {
+			versionFound = false
+		} else {
+			return objInfo, gerr
+		}
+	}
 
 	if opts.Expiration.Expire {
-		goi, _, err := er.getObjectInfoAndQuorum(ctx, bucket, object, opts)
-		if err == nil {
+		if gerr == nil {
 			evt := evalActionFromLifecycle(ctx, *lc, rcfg, goi)
 			var isErr bool
 			switch evt.Action {
@@ -1584,15 +1595,12 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 					Object: object,
 				}
 			}
-			markDelete = goi.VersionID != ""
 		}
 	}
 
-	versionFound := !(opts.DeleteMarker && opts.VersionID != "")
+	//  Determine whether to mark object deleted for replication
+	markDelete := goi.VersionID != ""
 
-	if !markDelete {
-		markDelete = !opts.DeleteMarker && opts.VersionID != ""
-	}
 	// Default deleteMarker to true if object is under versioning
 	// versioning suspended means we add `null` version as
 	// delete marker, if its not decided already.
@@ -1628,11 +1636,10 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 			ExpireRestored:   opts.Transition.ExpireRestored,
 		}
 		fi.SetTierFreeVersionID(fvID)
-		if opts.Versioned {
+		if opts.VersionID != "" {
+			fi.VersionID = opts.VersionID
+		} else if opts.Versioned {
 			fi.VersionID = mustGetUUID()
-			if opts.VersionID != "" {
-				fi.VersionID = opts.VersionID
-			}
 		}
 		// versioning suspended means we add `null` version as
 		// delete marker. Add delete marker, since we don't have

